launch_testing/launch_testing/test_result.py

```diff
@@ -33,6 +33,32 @@ def wasSuccessful(self):
         return False
 
 
+class SkipResult(unittest.TestResult):
+    """For test runs with a skip decorator on the generate_test_description function."""
+
+    class __skipped_test_case:
+
+        def __init__(self):
+            self._testMethodName = 'skipped_launch'
```

Comment:
@pbaughman meta: doesn't this make the whole XML output rather obscure? It removes all references to the actual tests being skipped.

Reply:
I don't know about 'rather obscure.'  The name of the test file is still in the XML output.  Your point is received, though. . .

Upon looking at this a second time, the test run is available at the point we create the SkipResult.  We could pass it into the `SkipResult` constructor and generate a full XML report that lists each test case as skipped.  That's what pytest does if you put a skip decorator on the test class.